### ISSUE TITLE
Header level and Style should not be directly and unbreakable connected

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5110]
Styles and Header level are not the same, an H1 should not be connected to a specific style.
Lexicon does not determinants this relationship, Lexicon just defines a set of styles:
https://liferay.design/lexicon/foundations/typography/#headings

Currently in heading component we can define the level and the label has associated a style ´text-´.
https://clayui.com/docs/components/heading.html

Initially I thought it could be changed by ´className´ but is not possible.

**Proposal:**
If we add an API new customization to change the style.. it could work, and the default style could persist.

**Comment by marcoscv-work** [comment]
On the other hand and just as a general question.. could be possible to know the order of all heading used in the page? in this way we could auto-order headings and prevent accessibility bad agrupations

For example: 
If we have this markup:

```
div.container
  h1.text-10
  h3.text-9
```

Autofix by remplace the wrong **h3** by the next **h2** keeping the styles as the designer wanted:

```
div.container
  h1.text-10
  h2.text-9
```

Of course, this could be deactivated :-)

**Comment by matuzalemsteles** [comment]
> Styles and Header level are not the same, an H1 should not be connected to a specific style.
Lexicon does not determinants this relationship, Lexicon just defines a set of styles:

We had these discussions in issue #4659 and #4604, there is more information because we associate the header level with the classes, we still have an old scale that was inherited from bootstrap but we will remove it in the next major release.

But I think this is a good point in a way but I also wonder why the h1 level could have different sizes? wouldn't that be inconsistent? I didn't want to allow modifying the `className` because that makes things more flexible allowing you to put anything in there and makes things more inconsistent, so maybe just an API to modify the component tag.

> On the other hand and just as a general question.. could be possible to know the order of all heading used in the page? in this way we could auto-order headings and prevent accessibility bad agrupations

[I have exactly this idea](https://github.com/liferay/clay/pull/4815#pullrequestreview-956773971) (This is the same idea that [Facebook did in its redesign](https://engineering.fb.com/2020/07/30/web/facebook-com-accessibility/)), to have this we would need to have a global Provider that manages this, we have the ClayProvider that could centralize the information but Liferay has multiple React trees being rendered separately, so I would have to use another strategy to store the information, it also gets harder when the DXP isn't all done in React, so we can't make assumptions from other parts of Liferay to tell which level is correct, this makes things more complicated.

**Comment by pat270** [comment]
My opinion is we should be less strict so Clay can be used by everyone, even though the main consumer is DXP. I believe Lexicon is trying to give designers more freedom to do what they want and I think this issue fits in with their vision. The [Lexicon site](https://liferay.design/lexicon/foundations/typography/) was updated and I don't see any references for `h1` - `h6` styles. This makes more sense because sometimes I want a smaller font for `h2` that still falls within the type scale approved by Lexicon.

@drakonux @emiliano-cicero

**Comment by matuzalemsteles** [comment]
Yeah, I think this is valid anyway I just want to make sure it doesn't create an inconsistency in the future, the less rigid we are the more the visual system has the freedom to create inconsistency due to teams not agreeing to reach a visual state consensus and that creates more variants for each use case...

This worries me, freedom for the design system is good for sure, we can't block creativity but also everyone should reflect the changes when something changes in the system, that's the great power of the design system, I think a great example of a well used design system is Apple's with IOS, it doesn't mean you block designers from evolving but if a change is made it is reflected everywhere and doesn't create too many variations of it.

My point here is that we have heading `h1` with different sizes depending on where it is being used but I understand that on some pages the size of `h1` may be different due to the context of the page, just confirming here.

Surely having something that sets that automatically is the next evolution of that and that would be great.

**Comment by pat270** [comment]
Yeah there will be no consistency by letting users pass in whatever they want, but we have a set of defaults that will be consistent.

**Comment by matuzalemsteles** [comment]
@marcoscv-work yeah, I understand. Ok I think we can add an API here that at least can change the font size, I thought I'd have an API like `as` to set the component tag to `<Heading as="h1" level={4} />` but it seems semantically wrong for this component, so maybe it makes more sense and seems more common to have a `fontSize` with scale levels `<Heading level={1} fontSize={4} />`.

I didn't want to add `className` just because it's an API that allows developers to do anything and change component semantic and get away from scaling for example, let me know what you think! I like this because the scale can be changed via sass variables if necessary so the whole system works fine.

**Comment by marcoscv-work** [comment]
How is this going? :-D

For the API use semantically what you think is better, fontSize can work :-D
